banyand/stream/stream_query_test.go

```diff
@@ -0,0 +1,532 @@
+					elements: []string{"1"},
+				},
+				{
+					id:       0,
+					location: []string{"series_1671844747554927007", "data_flow_0"},
+				},
+				{
+					id:       1,
+					location: []string{"series_2374367181827824198", "data_flow_0"},
+					elements: []string{"3"},
+				},
+				{
+					id:       1,
+					location: []string{"series_8429137420168685297", "data_flow_0"},
+				},
+			},
+		},
+		{
+			name: "order by duration",
+			args: queryOpts{
+				entity:    tsdb.Entity{tsdb.AnyEntry, tsdb.AnyEntry, tsdb.AnyEntry},
+				timeRange: tsdb.NewTimeRangeDuration(baseTime, 1*time.Hour),
+				buildFn: func(builder tsdb.SeekerBuilder) {
+					builder.OrderByIndex(&databasev2.IndexRule{
+						Metadata: &commonv2.Metadata{
+							Name:  "duration",
+							Group: "default",
+						},
+						Tags:     []string{"duration"},
+						Type:     databasev2.IndexRule_TYPE_TREE,
+						Location: databasev2.IndexRule_LOCATION_SERIES,
+					}, modelv2.QueryOrder_SORT_ASC)
+				},
+			},
+			want: shardsForTest{
+				{
+					id:       0,
+					location: []string{"series_12243341348514563931", "data_flow_0"},
+					elements: []string{"1"},
+				},
+				{
+					id:       0,
+					location: []string{"series_1671844747554927007", "data_flow_0"},
+					elements: []string{"2"},
+				},
+				{
+					id:       1,
+					location: []string{"series_2374367181827824198", "data_flow_0"},
+					elements: []string{"3", "5"},
+				},
+				{
+					id:       1,
+					location: []string{"series_8429137420168685297", "data_flow_0"},
+					elements: []string{"4"},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := queryData(tester, s, tt.args)
+			if tt.wantErr {
+				tester.Error(err)
+				return
+			}
+			tester.NoError(err)
+			sort.SliceStable(got, func(i, j int) bool {
+				a := got[i]
+				b := got[j]
+				if a.id > b.id {
+					return false
+				}
+				for i, al := range a.location {
+					bl := b.location[i]
+					if bytes.Compare([]byte(al), []byte(bl)) > 0 {
+						return false
+					}
+				}
+				return true
+			})
+			tester.Equal(tt.want, got)
+		})
+	}
+
+}
+
+func Test_Stream_Global_Index(t *testing.T) {
+	tester := assert.New(t)
+	s, deferFunc := setup(tester)
+	defer deferFunc()
+	_ = setupQueryData(tester, "global_index.json", s)
+	tests := []struct {
+		name                string
+		traceID             string
+		wantTraceSegmentNum int
+		wantErr             bool
+	}{
+		{
+			name:                "trace id is 1",
+			traceID:             "1",
+			wantTraceSegmentNum: 2,
+		},
+		{
+			name:                "trace id is 2",
+			traceID:             "2",
+			wantTraceSegmentNum: 3,
+		},
+		{
+			name:    "unknown trace id",
+			traceID: "foo",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			shards, errShards := s.Shards(nil)
+			tester.NoError(errShards)
+			err := func() error {
+				for _, shard := range shards {
+					itemIDs, err := shard.Index().Seek(index.Field{
```

Comment:
global index means it is directly attached to a single shard, right?

Reply:
it belongs to a segment in a shard. It means the global indices could be updated from time to time. 

Segment and block are the internal objects, query module could query a segment-attracted index, the global index, from a shard.